framework/src/executioners/Transient.C

```diff
@@ -344,8 +344,10 @@ Transient::execute()
      */
     if (!_fixed_point_solve->autoAdvance())
     {
+      _problem.finishMultiAppStep(EXEC_FIXED_POINT_BEGIN, /*recurse_through_multiapp_levels=*/true);
```

Comment:
this just makes sure that this 'finish' step is performed before we move on to the next step.

the finish step looks at the time steps iirc

Reply:
I think we want to put this in the outer solve object or executor. Sorry I added this comment in a wrong place that could misleading.

Reply:
The outer solve object might not need this though, there might be no fixed point iteration in which case we dont need this. 

I think I want to rename this execution flag to MULTIAPP_FIXED_POINT_BEGIN

Reply:
My main reason on this is that to minimize the number of execute flags we add. With these two new flags, essentially the fixed point iteration solve object is using four execute flags. If we move the two new one into timestepping, we do not have this issue.

Reply:
You use this PR to close https://github.com/idaholab/moose/issues/15611, which is about actual time step end. `FixedPointSolveObject` is one piece of the entire executioner, I have doubts on adding execute flags entering/exiting this solver object. The same functionality can be achieved by the caller who is calling `inner_solve->solve()` where the `inner_solve` is a fixed point solve object.

Reply:
This will indeed serve as an actual time step end, though by "actual time step end" you mean after the multiapp iteration and things, so really the fixed point end. So this name is fine. 

I think if you have nested executers/solve objects, then you want execution flags at the beginning and end of them.
This is already in the requirements for the executor system.

However in the mean time, we should be able to run on app in the fixed point iteration and one app once per fixed point iteration, and this is what it does.

Reply:
By timestepping, I mean adding the execute calls in a place like `TimeStepper::step()`. But since timestepper is currently entangled with `Transient`, I am not sure where the calls should be put. Somewhere in `Transient::execute()` might be more appropriate.

Reply:
Yeah, then this should not be used for closing #15611. And then back to my other comment about four execute flags used by one solve object. Why cannot we add the execute in `Transient::execute` or `TimeSteper::step`?

Reply:
We could. But it seems odd since neither of these are directly related to fixed point iterations.
We would be executing things on fixed_point_begin even though it's really timestep_begin at another location

Reply:
Yeah we cant. 

What I have here allows for the proper work of:
-backup / restore of multiapps
-failing the timestep if the multiapp on fixed_point_begin/end fail. This is trickier if outside of the FP iteration loop

Reply:
But I agree now that it wont do for #15611 since you have nested the fixed point iteration inside another executioner. So the end /begin of the FP is no longer the end/begin of the full timestep. 

For fixing that one, I reckon we either need the executor system or something like these "actual_timestep_*"  flags.

Reply:
This is a new capability having no impact on current Griffin. Plus getting backup/restore/failure handling all right is difficult, so I do not want to block this if this is needed. But be reminded that this is going to make future refactoring of time steppers even harder possibly.